packages/babel-template/src/index.js

```diff
@@ -10,7 +10,100 @@ import * as t from "babel-types";
 let FROM_TEMPLATE = "_fromTemplate"; //Symbol(); // todo: probably wont get copied over
 let TEMPLATE_SKIP = Symbol();
 
-export default function (code: string, opts?: Object): Function {
+export default function (firstArg, ...rest) {
+  if (typeof firstArg === "string") {
+    return factory(firstArg, ...rest);
+  } else {
+    return template(firstArg, ...rest);
+  }
+}
+
+function template (partials: Object | string[], ...args: Array<Object>) {
+  if (!Array.isArray(partials)) {
+    // support template({ options })`string`
+    return templateApply.bind(undefined, partials);
+  }
+  return templateApply(null, partials, ...args);
+}
+
+function templateApply(opts: Object | null, partials: string[], ...args: Array<Object>) {
+  if (partials.some((str) => str.includes("$BABEL_TEMPLATE$"))) {
+    throw new Error("Template contains illegal substring $BABEL_TEMPLATE$");
+  }
+
+  if (partials.length == 1) {
+    return factory(partials[0], opts);
+  }
+
+  const replacementSet = new Set();
+  const replacementMap = new Map();
+  const replacementValueMap = new Map();
+  let hasNonNumericReplacement = false;
+  for (const arg of args) {
+    if (replacementMap.has(arg)) {
+      continue;
+    }
+
+    if (typeof arg === "number") {
+      replacementMap.set(arg, `$${arg}`);
```

Comment:
This is using the current implementation's positional argument replacement.

It is validated that `0` cannot be used if named arguments are used too, but I forgot to check for `/$\d+/` in the string parts themselves...